src/vm/moar/QAST/QASTCompilerMAST.nqp

```diff
@@ -2614,6 +2631,67 @@ class MoarVM::BytecodeWriter {
         self.write_annotations;
         self.align;
     }
+    method assemble_to_file($filename) {
+        my $file := nqp::open($filename, 'w');
+        nqp::closefh($file);
+        $file := nqp::open($filename, '+');
+
+        # write the header, but we'll have to go back and correct some of the sizes
```

Comment:
Why don't you skip the header and write it last? It's always 92 bytes + 4 bytes padding long, so skipping is easy. That would get rid of the scary looking update code with those magic numbers further down.